Chezmoi breaks if config file has color: auto

## Describe the bug

CM throws an error if the config file has the option `color` set in `auto`

## To reproduce

Add the option and try to do `chezmoi apply` or `chezmoi doctor`

## Expected behavior

No error is thrown. I suggest to fail silently or handle the case.

## Output of command with the `--verbose` flag

```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104ddabc4]

goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/internal/cmd.(*autoBool).Value(...)
	github.com/twpayne/chezmoi/v2/internal/cmd/autobool.go:57
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).persistentPreRunRootE(0x14000204000, 0x14000719400, 0x10584cae0, 0x0, 0x0, 0x1057bd3a0, 0x0)
	github.com/twpayne/chezmoi/v2/internal/cmd/config.go:1151 +0x2dd4
github.com/spf13/cobra.(*Command).execute(0x14000719400, 0x10584cae0, 0x0, 0x0, 0x14000719400, 0x10584cae0)
	github.com/spf13/cobra@v1.2.1/command.go:835 +0x3ec
github.com/spf13/cobra.(*Command).ExecuteC(0x140000e6000, 0x140000e6000, 0x0, 0x0)
	github.com/spf13/cobra@v1.2.1/command.go:974 +0x264
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.2.1/command.go:902
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).execute(0x14000204000, 0x14000032050, 0x1, 0x1, 0x0, 0x0)
	github.com/twpayne/chezmoi/v2/internal/cmd/config.go:878 +0x70
github.com/twpayne/chezmoi/v2/internal/cmd.runMain(0x10507023c, 0x5, 0x105071500, 0x28, 0x1050713c0, 0x14, 0x105070df0, 0x8, 0x14000032050, 0x1, ...)
	github.com/twpayne/chezmoi/v2/internal/cmd/cmd.go:260 +0x94
github.com/twpayne/chezmoi/v2/internal/cmd.Main(0x10507023c, 0x5, 0x105071500, 0x28, 0x1050713c0, 0x14, 0x105070df0, 0x8, 0x14000032050, 0x1, ...)
	github.com/twpayne/chezmoi/v2/internal/cmd/cmd.go:87 +0x54
main.main()
	github.com/twpayne/chezmoi/v2/main.go:23 +0xec
```

## Output of `chezmoi doctor`

```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104ddabc4]

goroutine 1 [running]:
github.com/twpayne/chezmoi/v2/internal/cmd.(*autoBool).Value(...)
	github.com/twpayne/chezmoi/v2/internal/cmd/autobool.go:57
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).persistentPreRunRootE(0x14000204000, 0x14000719400, 0x10584cae0, 0x0, 0x0, 0x1057bd3a0, 0x0)
	github.com/twpayne/chezmoi/v2/internal/cmd/config.go:1151 +0x2dd4
github.com/spf13/cobra.(*Command).execute(0x14000719400, 0x10584cae0, 0x0, 0x0, 0x14000719400, 0x10584cae0)
	github.com/spf13/cobra@v1.2.1/command.go:835 +0x3ec
github.com/spf13/cobra.(*Command).ExecuteC(0x140000e6000, 0x140000e6000, 0x0, 0x0)
	github.com/spf13/cobra@v1.2.1/command.go:974 +0x264
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.2.1/command.go:902
github.com/twpayne/chezmoi/v2/internal/cmd.(*Config).execute(0x14000204000, 0x14000032050, 0x1, 0x1, 0x0, 0x0)
	github.com/twpayne/chezmoi/v2/internal/cmd/config.go:878 +0x70
github.com/twpayne/chezmoi/v2/internal/cmd.runMain(0x10507023c, 0x5, 0x105071500, 0x28, 0x1050713c0, 0x14, 0x105070df0, 0x8, 0x14000032050, 0x1, ...)
	github.com/twpayne/chezmoi/v2/internal/cmd/cmd.go:260 +0x94
github.com/twpayne/chezmoi/v2/internal/cmd.Main(0x10507023c, 0x5, 0x105071500, 0x28, 0x1050713c0, 0x14, 0x105070df0, 0x8, 0x14000032050, 0x1, ...)
	github.com/twpayne/chezmoi/v2/internal/cmd/cmd.go:87 +0x54
main.main()
	github.com/twpayne/chezmoi/v2/main.go:23 +0xec
```

--- a/internal/cmd/autobool.go
+++ b/internal/cmd/autobool.go
@@ -11,17 +11,8 @@ import (
 
 type autoBool struct {
 	auto     bool
-	autoFunc func() (bool, error)
 	value    bool
 	valueErr error
-}
-
-// newAutoBool returns a new autoBool.
-func newAutoBool(autoFunc func() (bool, error)) *autoBool {
-	return &autoBool{
-		auto:     true,
-		autoFunc: autoFunc,
-	}
 }
 
 // Set implements github.com/spf13/pflag.Value.Set.
@@ -52,9 +43,9 @@ func (b *autoBool) Type() string {
 }
 
 // Value returns b's value, calling b's autoFunc if needed.
-func (b *autoBool) Value() (bool, error) {
+func (b *autoBool) Value(autoFunc func() (bool, error)) (bool, error) {
 	if b.auto {
-		b.value, b.valueErr = b.autoFunc()
+		b.value, b.valueErr = autoFunc()
 		b.auto = false
 	}
 	return b.value, b.valueErr

--- a/internal/cmd/config.go
+++ b/internal/cmd/config.go
@@ -52,16 +52,16 @@ type templateConfig struct {
 // A Config represents a configuration.
 type Config struct {
 	// Global configuration, settable in the config file.
-	SourceDirAbsPath chezmoi.AbsPath                 `mapstructure:"sourceDir"`
-	DestDirAbsPath   chezmoi.AbsPath                 `mapstructure:"destDir"`
-	Umask            fs.FileMode                     `mapstructure:"umask"`
-	Remove           bool                            `mapstructure:"remove"`
 	Color            autoBool                        `mapstructure:"color"`
 	Data             map[string]interface{}          `mapstructure:"data"`
-	Template         templateConfig                  `mapstructure:"template"`
-	UseBuiltinGit    autoBool                        `mapstructure:"useBuiltinGit"`
-	Pager            string                          `mapstructure:"pager"`
+	DestDirAbsPath   chezmoi.AbsPath                 `mapstructure:"destDir"`
 	Interpreters     map[string]*chezmoi.Interpreter `mapstructure:"interpreters"`
+	Pager            string                          `mapstructure:"pager"`
+	Remove           bool                            `mapstructure:"remove"`
+	SourceDirAbsPath chezmoi.AbsPath                 `mapstructure:"sourceDir"`
+	Template         templateConfig                  `mapstructure:"template"`
+	Umask            fs.FileMode                     `mapstructure:"umask"`
+	UseBuiltinGit    autoBool                        `mapstructure:"useBuiltinGit"`
 
 	// Global configuration, not settable in the config file.
 	cpuProfile    chezmoi.AbsPath
@@ -202,18 +202,54 @@ func newConfig(options ...configOption) (*Config, error) {
 	}
 
 	c := &Config{
-		bds:        bds,
-		fileSystem: vfs.OSFS,
-		homeDir:    userHomeDir,
-		Umask:      chezmoi.Umask,
+		// Global configuration, settable in the config file.
 		Color: autoBool{
 			auto: true,
 		},
+		Interpreters: defaultInterpreters,
+		Pager:        os.Getenv("PAGER"),
+		Template: templateConfig{
+			Options: chezmoi.DefaultTemplateOptions,
+		},
+		Umask: chezmoi.Umask,
 		UseBuiltinGit: autoBool{
 			auto: true,
 		},
-		Pager:        os.Getenv("PAGER"),
-		Interpreters: defaultInterpreters,
+
+		// Global configuration, not settable in the config file.
+		homeDir:       userHomeDir,
+		templateFuncs: sprig.TxtFuncMap(),
+
+		// Password manager configurations, settable in the config file.
+		Bitwarden: bitwardenConfig{
+			Command: "bw",
+		},
+		Gopass: gopassConfig{
+			Command: "gopass",
+		},
+		Keepassxc: keepassxcConfig{
+			Command: "keepassxc-cli",
+		},
+		Lastpass: lastpassConfig{
+			Command: "lpass",
+		},
+		Onepassword: onepasswordConfig{
+			Command: "op",
+		},
+		Pass: passConfig{
+			Command: "pass",
+		},
+		Vault: vaultConfig{
+			Command: "vault",
+		},
+
+		// Encryption configurations, settable in the config file.
+		AGE: defaultAGEEncryptionConfig,
+		GPG: defaultGPGEncryptionConfig,
+
+		// Password manager data.
+
+		// Command configurations, settable in the config file.
 		Add: addCmdConfig{
 			exclude:   chezmoi.NewEntryTypeSet(chezmoi.EntryTypesNone),
 			include:   chezmoi.NewEntryTypeSet(chezmoi.EntryTypesAll),
@@ -241,33 +277,8 @@ func newConfig(options ...configOption) (*Config, error) {
 				"{{ .Target }}",
 			},
 		},
-		Template: templateConfig{
-			Options: chezmoi.DefaultTemplateOptions,
-		},
-		templateFuncs: sprig.TxtFuncMap(),
-		Bitwarden: bitwardenConfig{
-			Command: "bw",
-		},
-		Gopass: gopassConfig{
-			Command: "gopass",
-		},
-		Keepassxc: keepassxcConfig{
-			Command: "keepassxc-cli",
-		},
-		Lastpass: lastpassConfig{
-			Command: "lpass",
-		},
-		Onepassword: onepasswordConfig{
-			Command: "op",
-		},
-		Pass: passConfig{
-			Command: "pass",
-		},
-		Vault: vaultConfig{
-			Command: "vault",
-		},
-		AGE: defaultAGEEncryptionConfig,
-		GPG: defaultGPGEncryptionConfig,
+
+		// Command configurations, not settable in the config file.
 		apply: applyCmdConfig{
 			exclude:   chezmoi.NewEntryTypeSet(chezmoi.EntryTypesNone),
 			include:   chezmoi.NewEntryTypeSet(chezmoi.EntryTypesAll),
@@ -334,11 +345,16 @@ func newConfig(options ...configOption) (*Config, error) {
 			recursive: true,
 		},
 
+		// Configuration.
+		fileSystem: vfs.OSFS,
+		bds:        bds,
+
+		// Computed configuration.
+		homeDirAbsPath: homeDirAbsPath,
+
 		stdin:  os.Stdin,
 		stdout: os.Stdout,
 		stderr: os.Stderr,
-
-		homeDirAbsPath: homeDirAbsPath,
 	}
 
 	for key, value := range map[string]interface{}{

--- a/internal/cmd/config.go
+++ b/internal/cmd/config.go
@@ -56,10 +56,10 @@ type Config struct {
 	DestDirAbsPath   chezmoi.AbsPath                 `mapstructure:"destDir"`
 	Umask            fs.FileMode                     `mapstructure:"umask"`
 	Remove           bool                            `mapstructure:"remove"`
-	Color            *autoBool                       `mapstructure:"color"`
+	Color            autoBool                        `mapstructure:"color"`
 	Data             map[string]interface{}          `mapstructure:"data"`
 	Template         templateConfig                  `mapstructure:"template"`
-	UseBuiltinGit    *autoBool                       `mapstructure:"useBuiltinGit"`
+	UseBuiltinGit    autoBool                        `mapstructure:"useBuiltinGit"`
 	Pager            string                          `mapstructure:"pager"`
 	Interpreters     map[string]*chezmoi.Interpreter `mapstructure:"interpreters"`
 
@@ -202,10 +202,16 @@ func newConfig(options ...configOption) (*Config, error) {
 	}
 
 	c := &Config{
-		bds:          bds,
-		fileSystem:   vfs.OSFS,
-		homeDir:      userHomeDir,
-		Umask:        chezmoi.Umask,
+		bds:        bds,
+		fileSystem: vfs.OSFS,
+		homeDir:    userHomeDir,
+		Umask:      chezmoi.Umask,
+		Color: autoBool{
+			auto: true,
+		},
+		UseBuiltinGit: autoBool{
+			auto: true,
+		},
 		Pager:        os.Getenv("PAGER"),
 		Interpreters: defaultInterpreters,
 		Add: addCmdConfig{
@@ -385,8 +391,6 @@ func newConfig(options ...configOption) (*Config, error) {
 		return nil, err
 	}
 	c.DestDirAbsPath = c.homeDirAbsPath
-	c.Color = newAutoBool(c.colorAutoFunc)
-	c.UseBuiltinGit = newAutoBool(c.useBuiltinGitAutoFunc)
 	c._import.destination = c.homeDirAbsPath
 
 	return c, nil
@@ -761,7 +765,7 @@ func (c *Config) destAbsPathInfos(sourceState *chezmoi.SourceState, args []strin
 func (c *Config) diffFile(path chezmoi.RelPath, fromData []byte, fromMode fs.FileMode, toData []byte, toMode fs.FileMode) error {
 	var sb strings.Builder
 	unifiedEncoder := diff.NewUnifiedEncoder(&sb, diff.DefaultContextLines)
-	color, err := c.Color.Value()
+	color, err := c.Color.Value(c.colorAutoFunc)
 	if err != nil {
 		return err
 	}
@@ -975,11 +979,11 @@ func (c *Config) newRootCmd() (*cobra.Command, error) {
 
 	persistentFlags := rootCmd.PersistentFlags()
 
-	persistentFlags.Var(c.Color, "color", "Colorize output")
+	persistentFlags.Var(&c.Color, "color", "Colorize output")
 	persistentFlags.VarP(&c.DestDirAbsPath, "destination", "D", "Set destination directory")
 	persistentFlags.BoolVar(&c.Remove, "remove", c.Remove, "Remove entries from destination directory")
 	persistentFlags.VarP(&c.SourceDirAbsPath, "source", "S", "Set source directory")
-	persistentFlags.Var(c.UseBuiltinGit, "use-builtin-git", "Use builtin git")
+	persistentFlags.Var(&c.UseBuiltinGit, "use-builtin-git", "Use builtin git")
 	for _, key := range []string{
 		"color",
 		"destination",
@@ -1153,7 +1157,7 @@ func (c *Config) persistentPreRunRootE(cmd *cobra.Command, args []string) error 
 		c.errorf("warning: %s: %v\n", c.configFileAbsPath, err)
 	}
 
-	color, err := c.Color.Value()
+	color, err := c.Color.Value(c.colorAutoFunc)
 	if err != nil {
 		return err
 	}

--- a/internal/cmd/config_test.go
+++ b/internal/cmd/config_test.go
@@ -95,7 +95,7 @@ func TestParseConfig(t *testing.T) {
 			}, func(fileSystem vfs.FS) {
 				c := newTestConfig(t, fileSystem)
 				require.NoError(t, c.execute([]string{"init"}))
-				actualColor, err := c.Color.Value()
+				actualColor, err := c.Color.Value(c.colorAutoFunc)
 				assert.NoError(t, err)
 				assert.Equal(t, tc.expectedColor, actualColor)
 			})

--- a/internal/cmd/diffcmd.go
+++ b/internal/cmd/diffcmd.go
@@ -39,7 +39,7 @@ func (c *Config) newDiffCmd() *cobra.Command {
 func (c *Config) runDiffCmd(cmd *cobra.Command, args []string) error {
 	sb := strings.Builder{}
 	dryRunSystem := chezmoi.NewDryRunSystem(c.destSystem)
-	color, err := c.Color.Value()
+	color, err := c.Color.Value(c.colorAutoFunc)
 	if err != nil {
 		return err
 	}

--- a/internal/cmd/initcmd.go
+++ b/internal/cmd/initcmd.go
@@ -132,7 +132,7 @@ func (c *Config) runInitCmd(cmd *cobra.Command, args []string) error {
 			return err
 		}
 
-		useBuiltinGit, err := c.UseBuiltinGit.Value()
+		useBuiltinGit, err := c.UseBuiltinGit.Value(c.useBuiltinGitAutoFunc)
 		if err != nil {
 			return err
 		}

--- a/internal/cmd/testdata/scripts/config.txt
+++ b/internal/cmd/testdata/scripts/config.txt
@@ -17,6 +17,8 @@ chezmoi data --config=$CHEZMOICONFIGDIR/chezmoi.yaml --format=yaml
 stdout 'sourceDir: .*/config2/source'
 
 -- home2/user/.config/chezmoi/chezmoi.toml --
+color = "auto"
 sourceDir = "/config/source"
 -- home2/user/.config/chezmoi/chezmoi.yaml --
+color: auto
 sourceDir: /config2/source

--- a/internal/cmd/updatecmd.go
+++ b/internal/cmd/updatecmd.go
@@ -42,7 +42,7 @@ func (c *Config) newUpdateCmd() *cobra.Command {
 }
 
 func (c *Config) runUpdateCmd(cmd *cobra.Command, args []string) error {
-	switch useBuiltinGit, err := c.UseBuiltinGit.Value(); {
+	switch useBuiltinGit, err := c.UseBuiltinGit.Value(c.useBuiltinGitAutoFunc); {
 	case err != nil:
 		return err
 	case useBuiltinGit:

